data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -108,13 +115,29 @@ int processSqsMessages() {
     private List<Message> getMessagesFromSqs() {
         try {
             final ReceiveMessageRequest receiveMessageRequest = createReceiveMessageRequest();
-            return sqsClient.receiveMessage(receiveMessageRequest).messages();
-        } catch (SqsException e) {
-            LOG.error("Error reading from SQS: {}", e.getMessage());
+            final List<Message> messages = sqsClient.receiveMessage(receiveMessageRequest).messages();
+            failedAttemptCount = 0;
+            return messages;
+        } catch (final SqsException e) {
+            retryWithBackoff(e);
             return Collections.emptyList();
         }
     }
 
+    private void retryWithBackoff(final SqsException sqsException) {
```

Comment:
This method is inaccurately named and does not retry but rather just applies a backoff.